registry.go

```diff
@@ -115,6 +121,14 @@ type SignatureService interface {
 	// Get retrieves all of the signature blobs for the specified digest.
 	Get(dgst digest.Digest) ([][]byte, error)
 
+	// Enumerate retrieves all signature digests for given manifest revision,
+	// if it exists.
+	Enumerate(dgst digest.Digest) ([]digest.Digest, error)
+
 	// Put stores the signature for the provided digest.
 	Put(dgst digest.Digest, signatures ...[]byte) error
+
+	// Delete removes a signature link from particular manifest revision, if
```

Comment:
My original idea was to delete a single link which may eventually be corrupted, but come to think of it, it indeed makes no sense to delete just one and keeping the others. I'll change this.